js/locale/model-locale-resolver.js

```diff
@@ -27,128 +27,70 @@ export class ModelLocaleResolverPlugin extends RipeCommonsPlugin {
+            suffixes: suffixes
         });
     }
 
     localeProperty(
         name,
         { type = null, locale = null, defaultValue = null, prefixes = [], suffixes = [] } = {}
     ) {
-        let value = [];
-        value = value.concat(prefixes);
-        name && value.push(`properties.${name}`);
-        type && name && value.push(`properties.${type}.${name}`);
-        value = value.concat(suffixes);
-        return this.localeModel(value, {
+        const ripe = this.ripeProvider.ripe;
+        return ripe.localeProperty(name, this.localePlugin, {
+            brand: this.brand,
+            model: this.model,
+            type: type,
             locale: locale,
-            defaultValue: defaultValue
+            defaultValue: defaultValue,
+            prefixes: prefixes,
+            suffixes: suffixes
         });
     }
 
-    localeModel(value, { locale = null, defaultValue = null, ...options } = {}) {
-        options = Object.assign(
-            {
-                locale: locale,
-                defaultValue: defaultValue
-            },
-            options
-        );
-        return this._toLocale(value, this.brand, this.model, options);
-    }
-
     _bind() {
         this.owner.bind("config", config => {
             const { brand, name } = config || {};
             this.brand = brand;
             this.model = name;
         });
     }
-
-    _toLocale(
-        value,
-        brand,
-        model,
-        {
-            locale = null,
-            defaultValue = null,
-            prefix = "builds",
-            fallback = false,
-            compatibility = true,
-            hack = false
-        } = {}
-    ) {
-        const ripe = this.ripeProvider.ripe;
-        const values = Array.isArray(value) ? value : [value];
-        const result = ripe.localeModel(value, this.localePlugin, {
-            brand: brand,
-            model: model,
-            locale: locale,
-            defaultValue: defaultValue,
-            prefix: prefix,
-            fallback: fallback,
-            compatibility: compatibility,
-            hack: hack
-        });
-
-        // if the localization was successful returns its result
-        if (result !== defaultValue) return result;
-
-        // if the localization was not successful but a default
-        // was defined, then returns it
-        if (defaultValue !== null) return result;
-
-        // otherwise run the localization process again using
-        // a fallback locale as the base for localization
-        const localeFallback = this.localePlugin.getLocaleFallback();
-        if (localeFallback !== locale) {
-            return this._toLocale(values, brand, model, {
-                locale: localeFallback,
-                defaultValue: defaultValue,
-                prefix: prefix,
-                fallback: fallback,
-                compatibility: compatibility,
-                hack: hack
-            });
-        }
-
-        return values[0];

```

Comment:
You can't delete this because the method in RIPE SDK (the one that already existed, not the one you added) doesn't have the fallback logic, so this has to replicate it.

The changes from this PR need to be something like: use RIPE SDK whenever possible and add fallback at the end.